GDJS/Runtime/pixi-renderers/runtimescene-pixi-renderer.ts

```diff
@@ -120,44 +148,204 @@ namespace gdjs {
       if (displayHitboxesAndSomePoints) {
         for (let i = 0; i < instances.length; i++) {
           const object = instances[i];
+          const id = object.id;
+          const layer = this._runtimeScene.getLayer(object.getLayer());
+          const cameraTopLeftX =
+            layer.getCameraX() - layer.getCameraWidth() / 2;
+          const cameraTopLeftY =
+            layer.getCameraY() - layer.getCameraHeight() / 2;
+
+          if (
+            (!object.isVisible() || !layer.isVisible()) &&
+            !showHiddenInstances
+          ) {
+            continue;
+          }
           const cameraCoords = layersCameraCoordinates[object.getLayer()];
           const rendererObject = object.getRendererObject();
           if (!cameraCoords || !rendererObject) {
             continue;
           }
 
+          // Create the structure to store the points in memory
+          if (!this._debugDrawRenderedObjectsPoints[id]) {
+            this._debugDrawRenderedObjectsPoints[id] = {
+              wasRendered: true,
+              points: { x: 0, y: 0 },
+            };
+          }
+
+          const renderedObjectTextPoints = this._debugDrawRenderedObjectsPoints[
+            id
+          ];
+
           // Draw hitboxes (sub-optimal performance)
           const hitboxes = object.getHitBoxes();
           for (let j = 0; j < hitboxes.length; j++) {
             // Note that this conversion is sub-optimal, but we don't care
             // as this is for debug draw.
             const polygon: float[] = [];
+            let polyPointX: float = 0;
+            let polyPointY: float = 0;
             hitboxes[j].vertices.forEach((point) => {
-              polygon.push(point[0]);
-              polygon.push(point[1]);
+              if (layer) {
+                polygon.push(point[0] - cameraTopLeftX);
+                polygon.push(point[1] - cameraTopLeftY);
+              }
             });
             debugDraw.fill.alpha = 0;
-            debugDraw.line.color = 0xe86868;
+            debugDraw.line.alpha = 0.5;
+            debugDraw.line.color = 0xff0000;
             debugDraw.drawPolygon(polygon);
           }
 
-          // Draw circle point
-          debugDraw.fill.alpha = 0.8;
-          debugDraw.line.color = 0x68e868;
-          debugDraw.fill.color = 0x68e868;
-          debugDraw.drawCircle(object.getDrawableX(), object.getDrawableY(), 3);
-
-          // Draw center point
-          debugDraw.fill.alpha = 0.8;
-          debugDraw.line.color = 0xe8e868;
-          debugDraw.fill.color = 0xe8e868;
-          debugDraw.drawCircle(
-            object.getDrawableX() + object.getCenterX(),
-            object.getDrawableY() + object.getCenterY(),
-            3
-          );
+          // Draw points
+          // Center point of the object, with camera movement
+          const centerPoint = {
```

Comment:
No need to allocate memory for an object, just store two variables `centerPointX` and `centerPointY`